### PR TITLE
Check Database & Auth0 functionality on Render. Display all form entries rather than just client_entries.

### PR DESCRIPTION
**Pull request by nianokia**
# Need help with any of the following:
[My Render Link](https://server-ehsk.onrender.com/)
### Check Functionality
* Verify that forms work and Auth0 allows you to sign in
### Combine 2 unrelated tables
* Display all form entries, not just client entries on [Admin List Entries page](https://github.com/nianokia/wc-consulting/blob/d3d243edd7e770aa4f7cea48a2cbe51a709736ad/client/src/components/AdminListEntries.jsx)
### Auth0 Functions
* Fix error on login with Auth0
   * receiving POST https://*{myAuth0Domain}*/oauth/token 401 (Unauthorized) error on Login
* Conditionally display login and logout button
* On login, redirect to the Admin List Entries page automatically
* Provide feedback on code logic, syntax, readability or input on following best practices. 

gif

### Clarifications:
* On login, I want the admin to be redirected to the Admin List entries page. Then on logout, I want the admin to be redirected to the home page. 
   * That's why the buttons are currently hard-coded in, rather than conditionally displayed.

### Current status of Pushed-up Files
* [AdminListEntries.jsx](https://github.com/nianokia/wc-consulting/blob/d3d243edd7e770aa4f7cea48a2cbe51a709736ad/client/src/components/AdminListEntries.jsx) : needs to list all entries, not just client
* [server.js](https://github.com/nianokia/wc-consulting/blob/d3d243edd7e770aa4f7cea48a2cbe51a709736ad/server/server.js) : needs to gather all data from client_entries and professional_entries table
* ***All other files are in a good state***

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -102,76 +114,66 @@ Professional_Entry.sync().then((data) => {
...
+// --- retrieve all data from client_entries table ---
 app.get("/list", async (req, res) => {
     try {
         const client_entries = await Client_Entry.findAll();
         res.json(client_entries);
     } catch (err) {
         console.error("Error: ", err);
+        return res.status(400).json({ err });
     }
 });
```
Comment (tati-guzman) [inline server/server.js:151]: Within this try block, you can include a `professional_entries = await Professional_Entry.findAll()` to pull the data from your other table and send an object with { client_entries, professional_entries } after converting both to json. You'll have to update the mapping you do in the Admin List component to specify which part of the object you're pulling out, but it'll allow you to pull both at once for that component!

**Review by sunnybrie — COMMENTED** [review]
This is looking great! The only thing I'm finding is that the admin login is not redirecting to the admin list afterwards. (But the logout is redirecting as expected.)

**Review by tati-guzman — APPROVED** [review]
### General Feedback

You have so much figured out! It looks wonderful. The comments are clear and easy to follow to understand each file. I agree with the areas of improvement you've identified (Auth0 error and Admin List). Check out my comment in server.js for suggestions on pulling from both tables.

### Specifics

**Verify that forms work and Auth0 allows you to sign in**: 
- The forms are working great! I noticed that after adding myself as a client, I show up the client list correctly.
- Auth0 redirects correctly to its log in. I created an account and got a verification email and was able to use those credentials to continue signing in.
- Google Sign In: not quite working? Is it set up?

**Log In Redirection**
- Am also receiving the POST error
- Perhaps because of that error, it is not redirecting me to the Admin List page. BUT it won't let me try to log in again if I've "logged in" until I go to Admin List and Log Out.
- Redirect to homepage after log out works correctly

**Comment by nianokia** [comment]
> This is looking great! The only thing I'm finding is that the admin login is not redirecting to the admin list afterwards. (But the logout is redirecting as expected.)

## Login Redirect works locally but fails on Render

I got the redirect working locally! However on Render, it's only showing the list in JSON data, rather than the mapped list in AdminListEntries.jsx. The problem arose when I pushed these 3 commits: [79e1f58](https://github.com/nianokia/wc-consulting/pull/2/commits/79e1f587901fdda70433050de0c6bc2bce4f7520), [9a9f1bf](https://github.com/nianokia/wc-consulting/pull/2/commits/9a9f1bffa1bb405a7ccd4b7fe612af51eb373891), [4884cc6](https://github.com/nianokia/wc-consulting/pull/2/commits/4884cc66faeaa7d1587d21f6ff33ac900575b2d6).

I feel like the problem lies with the redirect to `${window.location.origin}/list` in the [handleLogin](https://github.com/nianokia/wc-consulting/blob/9470923cb48c1e07fe6d5a3215aecacda20d2b2f/client/src/components/Home.jsx#L16) function.

Here's a video of the problem.

https://github.com/user-attachments/assets/c31e4d8e-2f74-42fd-9169-5fdced70e187

### Any ideas on how to fix this?

File: server/server.js (hunk 1 above)
Comment (nianokia) [reply server/server.js:151]: Thank you so much for your review and for thoroughly testing the application! I didn't realize the Google sign in feature wasn't working universally, so I appreciate you pointing that out!

Also, I had the same idea to try to include a findAll for both tables, but it didn't work. However, I didn't update my mapping at the time, so I'll have to try it again with mapping this time! Good suggestion!

**Comment by nianokia** [comment]
## Login Redirect working both locally and in production

The problem was that my server was doing a [GET request ](https://github.com/nianokia/wc-consulting/blob/ccdc024a2ee0c269369aa0fc1b7c907297f25194/server/server.js#L143)for '/list' when it should've been routing to '/api/list' as defined in my proxy in [vite.config.js](https://github.com/nianokia/wc-consulting/blob/ccdc024a2ee0c269369aa0fc1b7c907297f25194/client/vite.config.js#L18)

https://github.com/user-attachments/assets/a6764f90-e180-4a70-ac50-8bbc69ac5685

**With this situated and working, I'm going to go ahead and merge it into main.** 🚀 

*I'll create a separate PR later to tackle gathering data from all tables successfully.*

**Pull request merged**